### ISSUE TITLE
gnt-job list can't filter with --pending with -o id

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#834]
Originally reported of Google Code with ID 778.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".
# gnt-cluster --version
gnt-cluster (ganeti v2.10.1) 2.10.1
# gnt-cluster version
Software version: 2.10.1
Internode protocol: 2100000
Configuration format: 2100000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.10.1
# hspace --version
hspace (ganeti) version v2.10.1
compiled with ghc 6.12
running on linux x86_64

<b>What distribution are you using?</b>
Debian squeeze

<b>What steps will reproduce the problem?</b>
1. gnt-job list --pending -o id

<b>What is the expected output? What do you see instead?</b>
Expected: A list of ID numbers that are the same as those listed by 'gnt-job list --pending'
Result: A list of many (all unarchived?) job IDs

<b>Please provide any additional information below.</b>
It appears to be somewhat specific to -o id. For instance, gnt-job list --pending -o id,status or just -o status returns a small number of jobs, even though -o id returns a large number.
```
Originally added on 2014-03-26 01:45:57 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
reproduced the problem
```
Originally added on 2014-03-26 09:55:47 +0000 UTC.

Added Labels: Type-Defect Priority-High
Changed State: Accepted
Added to Milestone: Release2.10

**Comment by Ganeti-Issues-Migrator** [comment]
```
Note that the problem exists in 2.10, but not on master. In needs to be fixed for the next 2.10 release nevertheless, of course.
```
Originally added on 2014-03-26 09:56:44 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-03-26 10:00:25 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I knew, I fixed that bug already... I believe, cherry-picking a8551d9c solves the problem.
```
Originally added on 2014-03-26 10:27:36 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I believe that this is fixed now

commit 0828635b9abffe6d203fec853f67a11a03818a78
Author: Klaus Aehlig <aehlig@google.com>
Date:   Tue Jan 7 18:27:37 2014 +0100

    Also consider filter fields for deciding if using live data
    
    If the query fields don't require live data, we use the shortcut
    and don't request live data. However, we cannot take this shortcut
    if the fields the filter depends on requires live data.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>
    
    Cherry-picked from a8551d9c.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>

```
Originally added on 2014-03-26 10:35:40 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released in 2.10.3.
```
Originally added on 2014-04-16 10:37:20 +0000 UTC.

Changed State: Released
